HW04.Rmd

```diff
@@ -1,126 +1,306 @@
+
+
+```{r month, warning=FALSE}
+
+deaths_per_month <- gun_deaths %>%
+    group_by(month) %>%
+    count()
+
+kable(deaths_per_month, format = "markdown",
+      col.names = c("Month", "Number of Deaths"),
+      caption = "Number of gun related deaths per month.")
+
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+#create plot
+ggplot(gun_deaths, aes(month)) +
+    geom_bar(fill = "cornflowerblue") +
+    scale_y_continuous(breaks = seq(0, 9000, by = 1000)) +
+    
+#add labels
+    labs(title = "Gun Deaths per Month from 2012-2014",
+         x     = "Month",
+         y     = "Number of Deaths") +
+    
+ #add theme   
+    theme_classic()
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+#create plot
+    #fct_infreq(): orders by number of observations with each level (largest first)
+ggplot(gun_deaths, aes(fct_infreq(intent))) +
+    geom_bar(fill = "cornflowerblue") +
+    scale_y_continuous(breaks = seq(0, 60000, by = 10000)) +
+    
+#add labels
+    labs(title = "Intent of Gun Related Deaths from 2012-2014",
+         x     = "Intent",
+         y     = "Number of Deaths") +
+    
+ #add theme   
+    theme_classic()
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+#create new data frame to remove NA age
+gun_deaths_by_sex <- gun_deaths %>%
+    filter(age != "Not Available") %>%
+    mutate(age = as.numeric(age))
+
+# generating a boxplot visualizing the data
+ggplot(gun_deaths_by_sex, aes(sex, age)) +
+    geom_boxplot(fill = "cornflowerblue") +
+    scale_y_continuous(breaks = seq(0, 100, by = 10)) +
+
+#add labels
+    labs(title = "Gun Related Deaths by Sex from 2012-2014",
+         x     = "Sex",
+         y     = "Age (years)") +
+    
+ #add theme   
+    theme_classic()
+
+
+#find average age of female gun death victims and create a new data frame
+female_avg_age <- gun_deaths %>%
+    group_by(sex) %>%
+    filter(sex == 'F') %>%
+    mutate(age = as.numeric(age)) %>%
+    summarize(average_age_female = mean(age, na.rm = TRUE)) # na.rm removes NA values
+
+#print out using kable()
+kable(female_avg_age, format = "markdown",
+      col.names = c("Sex", "Average age of Deaths"),
+      caption = "Average age of all female gun related deaths from 2012-2014")
+```
+
+
+## How many white males with at least a high school education were killed by guns in 2012?
+
+```{r education}
+#create new data frame
+white_m_hs_ed <- gun_deaths %>%
+    filter(sex == "M", 
+           race == "White", 
+           education %in% c('HS/GED', 'Some college', 'BA+'),
```

Comment:
Nice use of filtering!
!= could be used to negate 'Less than HS' to save a little typing, but I appreciate the improved readability your method offers.